riglib/ecube/__init__.py

```diff
@@ -382,103 +392,94 @@ def get(self):
     
 class LFP_Blanking(LFP_Plus_Trigger):
     '''
-    Blanks LFP data in the interval when the trigger is on. Compatible with riglib.source.MultiChanDataSource
+    Blanks LFP data in the interval when the trigger is on. Compatible with riglib.source.MultiChanDataSource.
+
+    Limitation: only blanks one chunk at a time, so smaller chunks are better! But -- the analog panel stream
+    arrives in more frequent 22-sample packets; once the headstage packets are that size then there is no 
+    benefit to making the hs packets any smaller! 
+    
+    Warning: untested! Use at your own risk.
     '''
 
-    blanking = False
-    analog_buffer = np.nan*np.zeros((1000*30,))
+    blanking = 0
     headstage_buffer = []
-        
+    trig_buffer = []
+    
     def get(self):
         '''data
         Retrieve a packet from the server
         '''
         try:
-            if self.blanking:
-                chan, data = next(self.gen)
-                blank = self.headstage_buffer[:,chan-1] # copy the last good headstage data for this channel
-                return (chan, blank)
-            else:
-                return next(self.gen)        
+            return next(self.gen)        
         except StopIteration:
             data_block = self.conn.get() # in the form of (time_stamp, data_source, data_content)
             while data_block[1] != "Headstages": # new packet of trigger data
-                trig = data_block[2][::25]
-                n_trig = len(trig)
-                self.analog_buffer[:-n_trig] = self.analog_buffer[n_trig:]
-                self.analog_buffer[-n_trig:] = trig
-                median = np.nanmedian(self.analog_buffer)
-                std = np.nanstd(self.analog_buffer)
-                thr = median + 3*std
-                if np.max(trig) > thr:
-                    self.blanking = True
-                else:
-                    self.blanking = False
+                self.trig_buffer = np.concatenate((self.trig_buffer, np.squeeze(data_block[2])), axis=0)
                 data_block = self.conn.get() # in the form of (time_stamp, data_source, data_content)
 
-            self.gen = multi_chan_generator(data_block[2], self.channels, downsample=25)
+            if len(self.trig_buffer) == 0:
+                self.blanking = 1
+            else:
+                trig_chunk_volts = analog_voltsperbit*np.array(self.trig_buffer)
             if self.blanking:
-                chan, data = next(self.gen)
-                blank = self.headstage_buffer[:,chan-1] # copy the last good headstage data for this channel
-                return (chan, blank)
+                self.blanking -= 1
+            elif (trig_chunk_volts[-1] > TRIG_THRESH and trig_chunk_volts[0] < TRIG_THRESH):
+                self.blanking = 1
+            elif (trig_chunk_volts[-1] < TRIG_THRESH and trig_chunk_volts[0] > TRIG_THRESH):
+                self.blanking = 1
```

Comment:
This is true, though this line is 2 in `LFP_Blanking_File`?

Reply:
thanks. for future reference, this setting controls how many chunks will be blanked. if needed it can be set to any positive integer. i've reset it to 1 for now by default.